docs/tutorial2.md

```diff
@@ -109,18 +109,17 @@ We have already hooked into the on idle event for building SCVs, we can use that
 Refactor your OnUnitIdle function with the following.
 
 ```C++
-virtual void OnUnitIdle(const Unit& unit) final {
-    switch (unit.unit_type.ToType()) {
+virtual void OnUnitIdle(const Unit* unit) final {
+    switch (unit->unit_type.ToType()) {
         case UNIT_TYPEID::TERRAN_COMMANDCENTER: {
             Actions()->UnitCommand(unit, ABILITY_ID::TRAIN_SCV);
             break;
         }
         case UNIT_TYPEID::TERRAN_SCV: {
-            uint64_t mineral_target;
-            if (!FindNearestMineralPatch(unit.pos, mineral_target)) {
+            if (!FindNearestMineralPatch(unit->pos)) {
                 break;
             }
-            Actions()->UnitCommand(unit, ABILITY_ID::SMART, mineral_target);
+            Actions()->UnitCommand(unit, ABILITY_ID::SMART, FindNearestMineralPatch(unit->pos));
```

Comment:
It's minor, but can you cache the results from the call to FindNearestMineralPatch above so it's not called twice.